modules/javafx.graphics/src/main/java/com/sun/javafx/font/coretext/CTFontFile.java

```diff
@@ -44,25 +47,49 @@ class CTFontFile extends PrismFontFile {
     }
 
     private static class SelfDisposerRecord implements DisposerRecord {
-        private long cgFontRef;
+        private long fontRef;
 
-        SelfDisposerRecord(long cgFontRef) {
-            this.cgFontRef = cgFontRef;
+        SelfDisposerRecord(long fontRef) {
+            this.fontRef = fontRef;
         }
 
         @Override
         public synchronized void dispose() {
-            if (cgFontRef != 0) {
-                OS.CFRelease(cgFontRef);
-                cgFontRef = 0;
+            if (fontRef != 0) {
+                OS.CFRelease(fontRef);
+                fontRef = 0;
             }
         }
     }
 
+    private long ctFontRef = 0;
+    CTFontFile(String name, String filename, int fIndex, long fontRef) throws Exception {
+        super(name, filename, fIndex, false, false, false, false);
+
+        if (fontRef == 0) {
+           throw new InternalError("Zero fontref");
+        }
+        ctFontRef = fontRef;
+        Disposer.addRecord(this, new SelfDisposerRecord(ctFontRef));
+    }
+
     CTFontFile(String name, String filename, int fIndex, boolean register,
                boolean embedded, boolean copy, boolean tracked) throws Exception {
         super(name, filename, fIndex, register, embedded, copy, tracked);
 
+        // The super-class code that opens and reads the font can't handle font variations,
+        // as used by the macOS "System Font"
+        // So when we see a font from this family, we need to use the original name
+        // passed in here as the full name.
+        // This isn't robust against Apple changing the name of the font
+        // but I think it sufficient until we add support for font variations.
+
+        if (name != null) {
+            String family = getFamilyName();
+            if (family.equals("System Font")) {
```

Comment:
very minor: I'd do `System Font".equals(...)`

Reply:
I do as you say when I'm worried about NPE. Here I'd quite like a NPE if this is null .. because there'd be something we'd need to investigate